src/control/lib/spdk/src/nvme_control_common.c

```diff
@@ -164,6 +164,9 @@ clean_ret(struct ret_t *ret)
 		free(ret->ctrlrs);
 		ret->ctrlrs = cnext;
 	}
+
+	if (ret)
+		free(ret);
```

Comment:
_collect calls clean_ret in the failure case, so I think we can't do the free here. Could either add a free_ret wrapper that calls clean_ret and then does the free, or do a cleanup function in the Go code that adds the free.